README.txt
$Id: README.txt,v 1.15 2009/07/24 17:47:47 dww Exp $

The signup module allows users to sign up for nodes of any type.  For
each signup-enabled node, this module provides options for sending a
notification email to a selected email address upon a new user signup
(good for notifying event coordinators, etc.), and a confirmation
email to users who sign up.  When used on event-enabled nodes
(http://drupal.org/project/event) or nodes that have a CCK date field
(http://drupal.org/project/date), it can also send out reminder emails
to all signups X days before the start of the event (per node
setting), and automatically close signups X hours before their start
(general setting).  There are settings for restricting signups to
selected roles and content types.

There is support for both registered and anonymous users to sign up.
Both can receive confirmation and reminder emails, and registered
users are optionally able to cancel or edit their own signups and view
listings of their current signups.

There is extensive integration with the Views module
(http://druapl.org/project/views), and also support to embed the
signup form as a pane using panels (http://druapl.org/project/panels).

Some add-on modules that enhance the basic functionality are available
in the "modules" subdirectory, and there is a listing of other add-on
modules on the Signup project page (http://druapl.org/project/signup).

For installation instructions, see INSTALL.txt.

For instructions on upgrading, see UPGRADE.txt.

For information about theming, see theme/README.txt.

Send feature requests and bug reports to the issue tracking system for
the signup module: http://drupal.org/node/add/project-issue/signup.
For a list of future work, also see http://groups.drupal.org/node/5044.

This module was originally co-developed by Chad Phillips and Jeff
Robbins, and sponsored by Jeff Robbins.  It is now being maintained
by: Derek Wright (http://drupal.org/user/46549) a.k.a. "dww".


CHANGELOG.txt
$Id: CHANGELOG.txt,v 1.123 2008/12/19 01:01:32 dww Exp $

-----------
6.x-2.x-dev
-----------
The DRUPAL-6--1 branch was created somewhat prematurely.  There's no
6.x-2.* series at this time.  Please use the DRUPAL-6--1 branch, instead.


-----------
6.x-1.0-rc2
-----------
Bugs fixed since 6.x-1.0-rc1:
= #336255 by KarenS and dww: Fixed numerous bugs in usage of DateAPI
  that were breaking all time-based signup functionality for datestamp
  fields and causing problems for certain configurations of date or
  datetime fields depending on the timezone handling.  Using
  6.x-2.0-rc5 of DateAPI and the CCK Date field, the one configuration
  that doesn't fully work is a datestamp field that has no timezone
  handling at all.  With that configuration, you must use the latest
  Date code from the DRUPAL-6--2 branch (check the top of your copy of
  date/date.module and make sure the revision is at least
  "1.61.2.4.2.34" (2008-11-28) or newer).
= #338850 by dww: Fixed critical SQL bug ("Not unique table/alias: 'e'")
  on the signup administration page (admin/content/signup) for sites
  that have more than one event.module node type that is signup-enabled.
= #335190 by bkat and dww: Fixed bug where if you had a node enabled
  for signups and edited it to disable signups, the node would remain
  signup enabled.  The bug was introduced in 5.x-2.6 by #328822.
= #336442 by dww: Fixed bugs with the views filters "Signup:
  Enabled/Disabled" and "Signup: Confirmation: Enabled/Disabled".
= #336461 by dww: The "Signup: Reminder: Enabled/Disabled" views
  filter was broken since it was using the wrong filter handler.
= #336450 by dww: Fixed the UI for signup's Views filters.
= #336610 by dww: Fixed bug introduced in #329668 where node types that
  have an underscore in their name had broken links in the warning
  messages about CCK date fields being misconfigured.  Those warnings
  also didn't appear at all in some cases.  The bug was caused by
  core's bizarre handling of node_type vs. node-type in URLs.
= #340377 by dww: Fixed bug where the default value in the database
  for how many days in advance to send reminder emails did not match
  what the rest of the code was written to expect.  Reminders would
  not be sent until the admin/settings/signup page was saved.

Other changes since 6.x-1.0-rc1:
= #335563 by dww: Added support to control the relative ordering of
  the signup information embedded on nodes.  If the content.module
  (CCK) is enabled, "Signup information" is now exposed to the field
  management page (where there is a drag-and-drop interface to reorder
  everything) at admin/content/node-type/[node-type-name]/fields.  If
  CCK is disabled, there is a variable to manually control the weight.
  The variable is called "signup_info_node_weight_[node_type]" and can
  be defined in the $conf array in settings.php.
= #336129 by dww: Moved the signup node type settings out of the
  "Workflow" fieldset into a separate "Signup settings" fieldset.
= #341331 by dww: Improved database performance during cron runs by
  using INNER JOIN instead of LEFT JOIN for finding nodes of a given
  type that need to be auto-closed and/or have reminder emails sent.


-----------
6.x-1.0-rc1
-----------
Changes since 6.x-1.0-beta1:
= #330828 by dww: Split module code out into separate include files.


-------------
6.x-1.0-beta1
-------------
Contains all the features and bug fixes up to and including version
5.x-2.6.  Changes since 5.x-2.6:
= #222217 by dww: Initial port to the Druapl 6 core API.
  Related issues: #330838, #330840, #332968, #333264.
= #330820 by dww and duellj: Ported all views support to the views2 API.
  This changes how the 'View to embed for the signup user list' setting
  works, so any sites that are using a custom view for this will have to
  revisit the signup settings page and select a new value under the
  advanced settings fieldset.  signup_update_6000() removes the
  deprecated variables from the database.
= #330821 by dww: Fixed problems with the D6 CCK Date field support.
= #330824 by dww: Ported the Panels integration to the D6 Panels API.
= #330829 by dww: Split theme functions out into separate include files.
= #333185 by dww: Removed deprecated signup_build_signup_data() function.


-------
5.x-2.7
-------
Bugs fixed since 5.x-2.6:
= #336255 by KarenS and dww: Fixed numerous bugs in usage of DateAPI
  that were breaking all time-based signup functionality for datestamp
  fields and causing problems for certain configurations of date or
  datetime fields depending on the timezone handling.  Using 5.x-2.4
  of DateAPI and the CCK Date field, the one configuration that
  doesn't fully work is a datestamp field that has no timezone
  handling at all.  With that configuration, you must use the latest
  Date code from the DRUPAL-5--2 branch (check the top of your copy of
  date_api.module and make sure the revision is at least "1.39.2.50"
  (2008-11-28) or newer).
= #338850 by dww: Fixed critical SQL bug ("Not unique table/alias: 'e'")
  on the signup administration page (admin/content/signup) for sites
  that have more than one event.module node type that is signup-enabled.
= #335190 by bkat and dww: Fixed bug where if you had a node enabled
  for signups and edited it to disable signups, the node would remain
  signup enabled.  The bug was introduced in 5.x-2.6 by #328822.
= 336610 by dww: Fixed bug introduced in #329668 where node types that
  have an underscore in their name had broken links in the warning
  messages about CCK date fields being misconfigured.  Those warnings
  also didn't appear at all in some cases.  The bug was caused by
  core's bizarre handling of node_type vs. node-type in URLs.
= #340377 by dww: Fixed bug where the default value in the database
  for how many days in advance to send reminder emails did not match
  what the rest of the code was written to expect.  Reminders would
  not be sent until the admin/settings/signup page was saved.


signup.api.php
<?php
// $Id: signup.api.php,v 1.5 2009/09/19 01:42:52 dww Exp $


/**
 * @file
 * This file documents the hooks invoked by the Signup module.
 */

/**
 * Hook to define forms available to signups to insert into the signup form.
 *
 * Forms should be provided by callback functions as a FormAPI array.
 * The callback should have the following signature:
 *   function my_callback(&$form_state, $node, $signup_type = 'auth')
 * See signup_basic_form_form for an example.
 * The values submitted to the form elements defined by this form will be 
 * serialized and stored in the {signup_log} table as 'form_data'.
 *
 * @return
 *   An array of possible forms, keyed by a unique ID. Each value is itself an 
 *   array of data, with the following key-value pairs:
 *     - 'label': (required) The human-readable name of the form.
 *     - 'description': (required) Extra information about the form.
 *     - 'callback': (required) The name of a function.
 *
 * @see signup_basic_form_form.
 */
function hook_signup_form_info() {
  return array(
    'basic' => array(
      'label' => 'Basic form',
      'description' => 'Collects name and phone number.',
      'callback' => 'signup_basic_form_form',
    ),  
  );
}

/**
 * Hook to alter signup data before a signup is inserted or updated.
 *
 * @param $signup
 *   Reference to the fully-loaded signup object representing the signup.
 * @param $form_values
 *   Array of form values (if any) from the signup being inserted or updated.
 */
function hook_signup_data_alter(&$signup, $form_values) {
  // TODO
}

/**
 * Hook invoked when a signup is being canceled.
 *
 * At the time this hook is invoked the record about the signup in the
 * {signup_log} table still exists, but the node has already had its signup
 * total decremented.
 *
 * @param $node
 *   The fully-loaded node object that the signup is being canceled from.
 * @param $signup
 *   An object containing all the known information about the signup being
 *   canceled. Contains all the data from the {signup_log} row representing
 *   the canceled signup. See the schema definition for descriptions of each
 *   field and what they represent.
 *
 * @return
 *   Ignored.
 *
 * @see signup_cancel_signup()
 */
function hook_signup_cancel($signup, $node) {
  $info = array();
  $info[] = t('Signup ID: @sid', array('@sid' => $signup->sid));
  $info[] = t('Node ID: @nid', array('@nid' => $signup->nid));
  $info[] = t('User ID: @uid', array('@uid' => $signup->uid));
  $info[] = t('Email address for anonymous signup: @anon_mail', array('@anon_mail' => $signup->anon_mail));
  $info[] = t('Date/time when the signup was created: @signup_time', array('@signup_time' => $signup->signup_time));
  $form_data = unserialize($signup->form_data);
  $info[] = t('Custom signup form data: %signup_form_data', array('%signup_form_data' => theme('signup_custom_data_email', $form_data)));
  $info[] = t('Attendance record: %attended', array('%attended' => theme('signup_attended_text', $signup->attended)));
  $info[] = t('Slots consumed by this signup: @count_towards_limit', array('@co
unt_towards_limit' => $signup->count_towards_limit));

  drupal_set_message(theme('item_list', $info, t('Signup canceled for %node_title', array('%node_title' => $node->title))));
}

/**
 * Hook invoked after a signup has been inserted.
 *
 * @param $signup
 *   The fully-loaded signup object representing the new signup.
 */
function hook_signup_insert($signup) {
  // TODO
}

/**
 * Hook invoked after a signup has been updated.
 *
 * @param $signup
 *   The fully-loaded signup object representing the updated signup.
 */
function hook_signup_update($signup) {
  // TODO
}

/**
 * Hook invoked when a signup is being created to gather other signup data.
 *
 * This hook allows other modules to inject information into the custom signup
 * data for each signup.  The array is merged with the values of any custom
 * fields from hook_signup_form_info(), serialized, and stored in the
 * {signup_log} database table.
 *
 * @param $node
 *   Fully-loaded node object being signed up to.
 * @param $account
 *   Full-loaded user object who is signing up.
 *
 * @return
 *   Keyed array of fields to include in the custom data for this signup. The
 *   keys for the array are used as labels when displaying the field, so they
 *   should be human-readable (and wrapped in t() to allow translation).
 *
 * @see signup_sign_up_user()
 * @see hook_signup_form_info()
 */
function hook_signup_sign_up($node, $account) {
  return array(
    t('Node type') => node_get_types('name', $node->type),
    t('User created') => format_date($account->created),
  );
}


/**
 * Hook invoked whenever a node is reopened for signups.
 *
 * A node with signups closed could be reopened in two main cases: 1) someone
 * cancels a signup and the signup limit is no longer reached; 2) a signup
 * administrator manually re-opens signups.
 *
 * @param $node
 *   Fully-loaded node object that is now open for signups.
 *
 * @return
 *   Ignored.
 *
 * @see signup_open_signup()
 */
function hook_signup_open($node) {
  drupal_set_message(t('Duplicate message: signups are now open on %title.', array('%title' => $node->title)));
}


/**
 * Hook invoked whenever a node is closed for signups.
 *
 * Signups are closed in 3 main cases: 1) it is a time-based node and the
 * close-in-advance time has been reached (auto-close via cron); 2) the node
 * has a signup limit and the limit is reached; 3) a signup administrator
 * manually closes signups.
 *
 * @param $node
 *   Fully-loaded node object that is now closed for signups.
 *
 * @return
 *   Ignored.
 *
 * @see signup_close_signup()
 */
function hook_signup_close($node) {
  drupal_set_message(t('Duplicate message: signups are now closed on %title.', array('%title' => $node->title)));
}


/**
 * Hook invoked to see if signup information should be printed for a node.
 *
 * This hook is invoked whenever someone is viewing a signup-enabled node and
 * allows modules to suppress any signup-related output.  If any module's
 * implementation of this hook returns TRUE, no signup information will be
 * printed for that node.
 *
 * @param $node
 *   The fully-loaded node object being viewed.
 *
 * @return
 *   TRUE if you want to prevent signup information from being printed, FALSE
 *   or NULL if the information should be printed.
 *
 * @see _signup_needs_output()
 * @see _signup_menu_access()
 * @see signup_nodeapi()
 */
function hook_signup_suppress($node) {
  if ($node->nid % 2) {
    drupal_set_message(t('Signup information suppressed for odd node ID %nid.', array('%nid' => $node->nid)));
    return TRUE;
  }
}

/**
 * Hook invoked to control access to signup menu items.
 *
 * This hook is invoked to check access for signup menu items, in particular,
 * the signup-related tabs on signup-enabled nodes. If no value is returned
 * (NULL), the hook is ignored and the usual access logic is enforced via the
 * Signup module. If multiple modules return a value, the logical OR is used,
 * so if anyone returns TRUE, access is granted. If everyone returns FALSE,
 * access is denied (even if the Signup module would normally grant access).
 *
 * @param $node
 *   The fully-loaded node object where the menu items would be attached.
 * @param $menu_type
 *   String specifying what kind of menu item to test access for. Can be:
 *   'signup': the signup form
 *   'list': the signup attendee listing
 *   'admin': the signup administration tab
 *   'add': the signup administration tab to add other users (requires
 *          that signups are currently open on the given node).
 *   'broadcast': for the broadcast tab
 *
 * @return
 *   TRUE if you want to allow access to the requested menu item, FALSE if you
 *   want to deny access (although if another hook implementation returns
 *   TRUE, that will take precedence), or NULL if you don't care and want to
 *   let Signup module itself decide access based on its own logic.
 *
 * @see _signup_menu_access()
 */
function hook_signup_menu_access($node, $menu_type) {
  // For example, you might want to test that the current user is the
  // administrator of the organic group that a signup-enabled node belongs to,
  // in which case, you'd return TRUE here to give that user full signup
  // powers over events in their group.
}



js/admin.settings.js
/* $Id: admin.settings.js,v 1.6 2009/01/24 08:52:40 dww Exp $ */

/**
 * Conditionally show/hide settings based on the signup form location.
 *
 * If the signup form is being show on the node itself, it's wrapped
 * in a fieldset and there's a setting to control if that fieldset
 * should be collapsed.  Only show this setting if the form is going
 * to be displayed on the node itself.
 */
Drupal.behaviors.signupShowFormLocationSetting = function () {
  $('div.signup-form-location-radios input.form-radio').click(function () {
    // Simple part: Depending on the form location, hide/show the
    // collapsible fieldset setting.
    if (this.value == 'node') {
      $('div.signup-fieldset_collapsed-setting').show();
    }
    else {
      $('div.signup-fieldset_collapsed-setting').hide();
    }
  });
};

/**
 * Conditionally show/hide settings based on the signup user list setting.
 *
 * If the signup user list is going to be an embedded view, show the
 * view-related settings, otherwise, hide them.
 */
Drupal.behaviors.signupShowUserListViewSetting = function () {
  $('div.signup-display-signup-user-list-setting input.form-radio').click(function () {
    if (this.value == 'embed-view' || this.value == 'embed-view-tab') {
      $('div.signup-user-list-view-settings').show();
    }
    else {
      $('div.signup-user-list-view-settings').hide();
    }
  });
};

/**
 * Conditionally show/hide settings based on the signup admin list setting.
 *
 * If the administer signup user list is going to be an embedded view,
 * show the view-related settings, otherwise, hide them.
 */
Drupal.behaviors.signupShowAdminListViewSetting = function () {
  $('div.signup-display-signup-admin-list-setting input.form-radio').click(function () {
    if (this.value == 'embed-view') {
      $('div.signup-admin-list-view-settings').show();
    }
    else {
      $('div.signup-admin-list-view-settings').hide();
    }
  });
};


js/signup_edit_form.js
/* $Id: signup_edit_form.js,v 1.4 2009/09/20 22:25:50 dww Exp $ */

/**
 * On the signup edit form, add some bling to make users "confirm" the edit.
 *
 * When the form first loads, all the form elements are disabled
 * except for the 'Update signup' button.  When a user clicks that,
 * the other form elements are enabled, and the 'Update signup' button
 * text changes into 'Save changes'.
 */
Drupal.behaviors.enableSignupEditForm = function(context) {
  if (!Drupal.settings.signupEditFormErrors) {
    var $button = $('#edit-save', context).click(enableSave);
    var $form = $button.parents('form:first');
    var $form_elements = $form.find(':visible');
    var original_button_title = $button.val();

    $form_elements.attr('disabled', 'disabled');
    $button.attr('disabled', '').val(Drupal.t('Edit'));
  }
 
  function enableSave() {
    $form_elements.attr('disabled', '');
    $button.unbind('click', enableSave).val(original_button_title);
    return false;
  }
}



js/admin.content_types.js
/* $Id: admin.content_types.js,v 1.4 2009/04/15 22:25:38 dww Exp $ */

/**
 * Conditionally show or hide the signup date field setting.
 *
 * On a node type settings form, if the "Allow signups" radios set
 * set to 0 ('Disabled'), or the node type is event-enabled, then
 * hide the date field setting.  Otherwise, show it.
 */
Drupal.behaviors.signupShowDateField = function () {
  $('div.signup-node-default-state-radios input[type=radio], div.event-nodeapi-radios input[type=radio]').click(function () {
    var eventEnabled = false;
    var signupDisabled = true;
    $('div.event-nodeapi-radios input.form-radio').each(function() {
      if (this.checked && this.value != 'never') {
        eventEnabled = true;
      }
    });
    $('div.signup-node-default-state-radios input.form-radio').each(function() {
      if (this.checked && this.value != 'disabled') {
        signupDisabled = false;
      }
    });
    if (signupDisabled || eventEnabled) {
      $('div.signup-date-field-setting').hide();
    }
    else {
      $('div.signup-date-field-setting').show();
    }
  });
};


views/default_views/signup_user_list.view.php
<?php
// $Id: signup_user_list.view.php,v 1.2 2009/10/03 02:27:39 dww Exp $

/**
 * @file
 * All users who have signed up for a given node.
 */

$view = new view;
$view->name = 'signup_user_list';
$view->description = 'List of all users signed up for a given node';
$view->tag = 'Signup';
$view->view_php = '';
$view->base_table = 'signup_log';
$view->is_cacheable = FALSE;
$view->api_version = 2;
$view->disabled = FALSE;
$handler = $view->new_display('default', 'Defaults', 'default');
$handler->override_option('relationships', array(
  'uid' => array(
    'label' => 'Signup user',
    'required' => 1,
    'id' => 'uid',
    'table' => 'signup_log',
    'field' => 'uid',
    'relationship' => 'none',
  ),
  'nid' => array(
    'label' => 'Signup node',
    'required' => 1,
    'id' => 'nid',
    'table' => 'signup_log',
    'field' => 'nid',
    'relationship' => 'none',
  ),
));
$handler->override_option('fields', array(
  'name' => array(
    'label' => 'Name',
    'link_to_user' => 1,
    'exclude' => 0,
    'id' => 'name',
    'table' => 'users',
    'field' => 'name',
    'relationship' => 'uid',
  ),
  'signup_time' => array(
    'label' => 'Signup time',
    'date_format' => 'small',
    'custom_date_format' => '',
    'exclude' => 0,
    'id' => 'signup_time',
    'table' => 'signup_log',
    'field' => 'signup_time',
    'relationship' => 'none',
  ),
));
$handler->override_option('arguments', array(
  'nid' => array(
    'default_action' => 'not found',
    'style_plugin' => 'default_summary',
    'style_options' => array(),
    'wildcard' => 'all',
    'wildcard_substitution' => 'All',
    'title' => '',
    'default_argument_type' => 'fixed',
    'default_argument' => '',
    'validate_type' => 'signup_status',
    'validate_fail' => 'not found',
    'break_phrase' => 0,
    'not' => 0,
    'id' => 'nid',
    'table' => 'node',
    'field' => 'nid',
    'relationship' => 'nid',
    'default_options_div_prefix' => '',
    'default_argument_user' => 0,
    'default_argument_fixed' => '',
    'default_argument_php' => '',
    'validate_argument_node_type' => array(),
    'validate_argument_node_access' => 0,
    'validate_argument_nid_type' => 'nid',
    'validate_argument_vocabulary' => array(),
    'validate_argument_type' => 'tid',
    'validate_argument_signup_status' => 'any',
    'validate_argument_signup_node_access' => 1,
    'validate_argument_php' => '',
  ),
));
$handler->override_option('access', array(
  'type' => 'perm',
  'perm' => 'view all signups',
));
$handler->override_option('items_per_page', 100);
$handler->override_option('use_pager', 1);
$handler->override_option('style_plugin', 'table');
$handler->override_option('style_options', array(
  'grouping' => '',
  'override' => 1,
  'sticky' => 0,
  'order' => 'asc',
  'columns' => array(
    'name' => 'name',
    'signup_time' => 'signup_time',
  ),
  'info' => array(
    'name' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'signup_time' => array(
      'sortable' => 1,
      'separator' => '',
    ),
  ),
  'default' => 'signup_time',
));


views/default_views/signup_user_admin_list.view.php
<?php
// $Id: signup_user_admin_list.view.php,v 1.2 2009/10/03 02:27:39 dww Exp $

/**
 * @file
 * Administrative list of all users signed up for a given node.
 */

$view = new view;
$view->name = 'signup_user_admin_list';
$view->description = t('Administrative list of all users signed up for a given node.');
$view->tag = 'Signup';
$view->view_php = '';
$view->base_table = 'signup_log';
$view->is_cacheable = FALSE;
$view->api_version = 2;
$view->disabled = FALSE;
$handler = $view->new_display('default', 'Defaults', 'default');
$handler->override_option('relationships', array(
  'nid' => array(
    'label' => 'Signup node',
    'required' => 1,
    'id' => 'nid',
    'table' => 'signup_log',
    'field' => 'nid',
    'relationship' => 'none',
  ),
  'uid' => array(
    'label' => 'Signup user',
    'required' => 1,
    'id' => 'uid',
    'table' => 'signup_log',
    'field' => 'uid',
    'relationship' => 'none',
  ),
));
$handler->override_option('fields', array(
  'name' => array(
    'label' => 'Name',
    'link_to_user' => 1,
    'exclude' => 0,
    'id' => 'name',
    'table' => 'users',
    'field' => 'name',
    'relationship' => 'uid',
  ),
  'signup_time' => array(
    'label' => 'Signup time',
    'date_format' => 'small',
    'custom_date_format' => '',
    'exclude' => 0,
    'id' => 'signup_time',
    'table' => 'signup_log',
    'field' => 'signup_time',
    'relationship' => 'none',
  ),
  'form_data' => array(
    'label' => 'Additional info',
    'type' => 'ul',
    'separator' => ', ',
    'empty' => '',
    'exclude' => 0,
    'id' => 'form_data',
    'table' => 'signup_log',
    'field' => 'form_data',
    'relationship' => 'none',
    'form_data_fieldname' => '',
  ),
  'attended' => array(
    'label' => 'Attendance',
    'exclude' => 0,
    'id' => 'attended',
    'table' => 'signup_log',
    'field' => 'attended',
    'relationship' => 'none',
  ),
  'edit_link' => array(
    'label' => 'Operations',
    'text' => '',
    'exclude' => 0,
    'id' => 'edit_link',
    'table' => 'signup_log',
    'field' => 'edit_link',
    'relationship' => 'none',
  ),
));
$handler->override_option('arguments', array(
  'nid' => array(
    'default_action' => 'ignore',
    'style_plugin' => 'default_summary',
    'style_options' => array(),
    'wildcard' => 'all',
    'wildcard_substitution' => 'All',
    'title' => '',
    'default_argument_type' => 'fixed',
    'default_argument' => '',
    'validate_type' => 'signup_status',
    'validate_fail' => 'not found',
    'break_phrase' => 0,
    'not' => 0,
    'id' => 'nid',
    'table' => 'node',
    'field' => 'nid',
    'relationship' => 'nid',
    'default_options_div_prefix' => '',
    'default_argument_user' => 0,
    'default_argument_fixed' => '',
    'default_argument_php' => '',
    'validate_argument_node_access' => 0,
    'validate_argument_nid_type' => 'nid',
    'validate_argument_vocabulary' => array(),
    'validate_argument_type' => 'tid',
    'validate_argument_signup_status' => 'any',
    'validate_argument_signup_node_access' => 1,
    'validate_argument_php' => '',
  ),
));
$handler->override_option('access', array(
  'type' => 'none',
));
$handler->override_option('items_per_page', 100);
$handler->override_option('use_pager', 1);
$handler->override_option('style_plugin', 'table');
$handler->override_option('style_options', array(
  'grouping' => '',
  'override' => 1,
  'sticky' => 0,
  'order' => 'desc',
  'columns' => array(
    'name' => 'name',
    'signup_time' => 'signup_time',
    'form_data' => 'form_data',
    'attended' => 'attended',
    'edit_link' => 'edit_link',
  ),
  'info' => array(
    'name' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'signup_time' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'form_data' => array(
      'separator' => '',
    ),
    'attended' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'edit_link' => array(
      'separator' => '',
    ),
  ),
  'default' => 'signup_time',
));


views/default_views/signup_user_vbo_admin_list.view.php
<?php
// $Id: signup_user_vbo_admin_list.view.php,v 1.2 2009/10/03 02:27:39 dww Exp $

/**
 * @file
 * Views Bulk Operations (VBO) view for administering signups on a node.
 */

$view = new view;
$view->name = 'signup_user_vbo_admin_list';
$view->description = t('Per-node signup administration interface with checkboxes for bulk operations.');
$view->tag = 'Signup';
$view->view_php = '';
$view->base_table = 'signup_log';
$view->is_cacheable = FALSE;
$view->api_version = 2;
$view->disabled = FALSE;
$handler = $view->new_display('default', 'Defaults', 'default');
$handler->override_option('relationships', array(
  'nid' => array(
    'label' => 'Signup node',
    'required' => 1,
    'id' => 'nid',
    'table' => 'signup_log',
    'field' => 'nid',
    'relationship' => 'none',
  ),
  'uid' => array(
    'label' => 'Signup user',
    'required' => 1,
    'id' => 'uid',
    'table' => 'signup_log',
    'field' => 'uid',
    'relationship' => 'none',
  ),
));
$handler->override_option('fields', array(
  'name' => array(
    'label' => 'Name',
    'link_to_user' => 1,
    'exclude' => 0,
    'id' => 'name',
    'table' => 'users',
    'field' => 'name',
    'relationship' => 'uid',
  ),
  'signup_time' => array(
    'label' => 'Signup time',
    'date_format' => 'small',
    'custom_date_format' => '',
    'exclude' => 0,
    'id' => 'signup_time',
    'table' => 'signup_log',
    'field' => 'signup_time',
    'relationship' => 'none',
  ),
  'form_data' => array(
    'label' => 'Additional info',
    'type' => 'ul',
    'separator' => ', ',
    'empty' => '',
    'exclude' => 0,
    'id' => 'form_data',
    'table' => 'signup_log',
    'field' => 'form_data',
    'relationship' => 'none',
    'form_data_fieldname' => '',
  ),
  'attended' => array(
    'label' => 'Attendance',
    'exclude' => 0,
    'id' => 'attended',
    'table' => 'signup_log',
    'field' => 'attended',
    'relationship' => 'none',
  ),
  'edit_link' => array(
    'label' => 'Operations',
    'text' => '',
    'exclude' => 0,
    'id' => 'edit_link',
    'table' => 'signup_log',
    'field' => 'edit_link',
    'relationship' => 'none',
  ),
));
$handler->override_option('arguments', array(
  'nid' => array(
    'default_action' => 'ignore',
    'style_plugin' => 'default_summary',
    'style_options' => array(),
    'wildcard' => 'all',
    'wildcard_substitution' => 'All',
    'title' => '',
    'default_argument_type' => 'fixed',
    'default_argument' => '',
    'validate_type' => 'signup_status',
    'validate_fail' => 'not found',
    'break_phrase' => 0,
    'not' => 0,
    'id' => 'nid',
    'table' => 'node',
    'field' => 'nid',
    'relationship' => 'nid',
    'default_options_div_prefix' => '',
    'default_argument_user' => 0,
    'default_argument_fixed' => '',
    'default_argument_php' => '',
    'validate_argument_node_access' => 0,
    'validate_argument_nid_type' => 'nid',
    'validate_argument_vocabulary' => array(),
    'validate_argument_type' => 'tid',
    'validate_argument_signup_status' => 'any',
    'validate_argument_signup_node_access' => 1,
    'validate_argument_php' => '',
  ),
));
$handler->override_option('access', array(
  'type' => 'none',
));
$handler->override_option('items_per_page', 100);
$handler->override_option('use_pager', 1);
$handler->override_option('style_plugin', 'bulk');
$handler->override_option('style_options', array(
  'grouping' => '',
  'override' => 1,
  'sticky' => 0,
  'order' => 'asc',
  'columns' => array(
    'name' => 'name',
    'signup_time' => 'signup_time',
    'form_data' => 'form_data',
    'attended' => 'attended',
    'edit_link' => 'edit_link',
  ),
  'info' => array(
    'name' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'signup_time' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'form_data' => array(
      'separator' => '',
    ),
    'attended' => array(
      'sortable' => 1,
      'separator' => '',
    ),
    'edit_link' => array(
      'separator' => '',
    ),
  ),
  'default' => '-1',
  'execution_type' => '1',
  'display_type' => '0',
  'skip_confirmation' => 0,
  'display_result' => 1,
  'selected_operations' => array(
    '5c7373a6aa7a8bf5549412c9f887c2d3' => '5c7373a6aa7a8bf5549412c9f887c2d3',
    'd3a1a31a80ae6f3a3445aa18d827c62e' => 'd3a1a31a80ae6f3a3445aa18d827c62e',
    '486e69f274fc0c7fb96d00a55ce752ae' => '486e69f274fc0c7fb96d00a55ce752ae',
  ),
));
